### ISSUE TITLE
email addresses must be unique

### ISSUE THREAD

**Issue by engerm** [RagtagOpen/nomad#186]
There is an issue with allowing a users to change email address.

There are two scenarios that can happen:
1. locking a user out of their account
2. reducing a user's privilege (removing admin role)

The determining factor is which user id was created first.

Expected results:
1) Email addresses should be unique
2) An email would be sent to old/original account notifying them their email address has been changed
2b) The email sent to the old/original account would have link allowing them to directly log into their account to access their account and reset their email

**Comment by iandees** [comment]
The question here is what to do when someone tries to login with a social account whose email address already exists in the system.

1. Does the second login automatically merge with the first account and seamlessly log them in? Is there a security concern there?
2. Should we prevent the login and ask them to go login with their other social account?

**Comment by karpet** [comment]
@iandees those are good questions.

My suggestion, from a UX and security perspective, is *not* to automatically merge accounts, in the event that one of them has been compromised.

I would opt for number 2, and also add messaging and a feature to allow an already-logged in user to associate their account with a new 3rd party social account. This allows the user to merge their accounts only after they have verified they own the existing one.

**Comment by AaronSofaer** [comment]
Every website I've used that has both social oauth and password-based login prompts you to log in with your created account if there's a conflict. None of them does merges.

**Comment by sjohnson540** [comment]
I believe some sites ask if you want to merge _after_ you have logged in with the already created account on said website. I have seen this pattern on a few different sites. (which is what @karpet said)

**Comment by jillh510** [comment]
As it is right now, I was able to log in via FB and Google (same email address), which then appeared to make me two different accounts connected to the same email. I tested this by
- Logged in via FB, checked the Profile page and saw my display name was "Jill Huchital"
- Logged out
- Logged in via Google, checked the Profile page and saw my display name was "Jill Huchital"
- Changed my display name to "Giraffe"
- Logged out
- Logged in via FB, checked the Profile page and saw my display name was "Jill Huchital"
So I seem to have two profiles, connected to the same email address.

Let's change to option (2) - prevent the second login and prompt the user to log in via the other method. And I'm OK not dealing with merging at the moment.
